src/mbgl/renderer/layers/render_circle_layer.cpp

```diff
@@ -17,6 +17,32 @@ using namespace style;
 
 namespace {
 
+struct RenderableSegment {
+    RenderableSegment(const Segment<CircleAttributes>& segment_,
+                      CircleProgram& programInstance_,
+                      const CircleBucket& bucket_,
+                      const CircleProgram::UniformValues& allUniformValues_,
+                      const CircleProgram::AttributeBindings& allAttributeBindings_,
+                      float sortKey_)
+        : segment(segment_),
+          programInstance(programInstance_),
+          bucket(bucket_),
+          allUniformValues(allUniformValues_),
+          allAttributeBindings(allAttributeBindings_),
+          sortKey(sortKey_) {}
+
+    const Segment<CircleAttributes>& segment;
+    CircleProgram& programInstance;
+    const CircleBucket& bucket;
+    const CircleProgram::UniformValues& allUniformValues;
+    const CircleProgram::AttributeBindings& allAttributeBindings;
+    const float sortKey;
+
+    friend bool operator<(const RenderableSegment& lhs, const RenderableSegment& rhs) {
+        return lhs.sortKey < rhs.sortKey;
```

Comment:
sort by tile id if sort key is the same?
https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/layers/render_symbol_layer.cpp#L99